tasks_management/services.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+from django.db import transaction
+
+from core.services import create_or_update_core_user, BaseService
+from core.services.utils import check_authentication, output_exception, model_representation, output_result_success
+from tasks_management.models import TaskGroup, TaskExecutor
+from tasks_management.validation import TaskGroupValidation, TaskExecutorValidation
+
+logger = logging.getLogger(__name__)
+
+
+class TaskGroupService:
+    OBJECT_TYPE = TaskGroup
+
+    def __init__(self, user):
+        self.user = user
+        self.validation_class = TaskGroupValidation
+
+    @check_authentication
+    def create(self, obj_data):
+        try:
+            with transaction.atomic():
+                user_ids = obj_data.pop('user_ids')
+                self.validation_class.validate_create(self.user, **obj_data)
+                obj_ = self.OBJECT_TYPE(**obj_data)
+                task_group_output = self.save_instance(obj_)
+                task_group_id = task_group_output['data']['id']
+                task_executor_service = TaskExecutorService(self.user)
+                for user_id in user_ids:
+                    task_executor_service.create({
+                        "task_group_id": task_group_id,
+                        "user_id": user_id
+                    })
+                return task_group_output
+        except Exception as exc:
+            return output_exception(model_name=self.OBJECT_TYPE.__name__, method="create", exception=exc)
+
+    def save_instance(self, obj_):
+        obj_.save(username=self.user.username)
+        dict_repr = model_representation(obj_)
+        return output_result_success(dict_representation=dict_repr)
```

Comment:
Returning full object reflection seems to be inefficient. Is this a common approach in other places in this module? I think that uuid should be sufficient, however there might be cases when some data is used later on in the code and adding change in this place would have negative impact on the efficiency.

Reply:
Yes, this is how save_instance in BaseModel works. 
![image](https://github.com/openimis/openimis-be-tasks_management_py/assets/58431955/7b835428-142c-4bc2-bda5-5703b3955984)